Frends.AmazonS3.DownloadObject/Frends.AmazonS3.DownloadObject.Tests/UnitTest1.cs

```diff
@@ -0,0 +1,446 @@
+            Assert.IsTrue(ex.Message.Contains("Consider using Overwrite or ContinueIfExists options"));
+        }
+
+        /// <summary>
+        /// File already exists in destination. Overwrite = false and ContinueIfExists = true. Result will contain notification about the skipped file but continued downloading other files.
+        /// </summary>
+        [TestMethod]
+        public void AWSCredsContinueIfExistsTest()
+        {
+            _input = new Input()
+            {
+                AuthenticationMethod = AuthenticationMethod.AWSCredentials,
+                AwsAccessKeyId = _accessKey,
+                AwsSecretAccessKey = _secretAccessKey,
+                BucketName = _bucketName,
+                Region = Region.EuCentral1,
+                DestinationPath = @$"{_dir}\Download",
+                S3Directory = "2022/",
+                SearchPattern = "*",
+                ContinueIfExists = true,
+                Overwrite = false,
+                DeleteSourceFile = false,
+                DownloadFromCurrentDirectoryOnly = false,
+                ThrowErrorIfNoMatch = false,
+            };
+
+            var result = AmazonS3.DownloadObject(_input, default);
+            Assert.IsNotNull(result.Result.Results.Any(x => x.ObjectData.Contains("File Overwrite.txt was skipped because it already exists at ")));
+            Assert.IsTrue(File.Exists(@$"{_dir}\Download\Testfile.txt"));
+            Assert.IsTrue(File.Exists(@$"{_dir}\Download\DownloadFromCurrentDirectoryOnly.txt"));
+        }
+
+        /// <summary>
+        /// Download only from S3 2022/ folder. DownloadFromCurrentDirectoryOnly.txt shouldn't exists.
+        /// </summary>
+        [TestMethod]
+        public void AWSCredsDownloadFromCurrentDirectoryOnlyTest()
+        {
+            _input = new Input()
+            {
+                AuthenticationMethod = AuthenticationMethod.AWSCredentials,
+                AwsAccessKeyId = _accessKey,
+                AwsSecretAccessKey = _secretAccessKey,
+                BucketName = _bucketName,
+                Region = Region.EuCentral1,
+                DestinationPath = @$"{_dir}\Download",
+                S3Directory = "2022/",
+                SearchPattern = "*",
+                DownloadFromCurrentDirectoryOnly = true,
+                Overwrite = true,
+                DeleteSourceFile = false,
+                ThrowErrorIfNoMatch = false,
+            };
+
+            var result = AmazonS3.DownloadObject(_input, default);
+            Assert.IsNotNull(result.Result.Results);
+            Assert.IsTrue(File.Exists(@$"{_dir}\Download\Testfile.txt"));
+            Assert.IsTrue(File.Exists(@$"{_dir}\Download\Overwrite.txt"));
+            Assert.IsFalse(File.Exists(@$"{_dir}\Download\DownloadFromCurrentDirectoryOnly.txt"));
+        }
+
+        /// <summary>
+        /// Delete source files.
+        /// </summary>
+        /*
+        [TestMethod]
+        public void AWSCredsDeleteSourceFileTest()
```

Comment:
Please remove the code that is documented away.

Reply:
DeleteSourceFile is one of the options and I think it's important to show that it has been tested. I commented this one so that we don't have to upload files to S3 after each full tests / github's test. Uncommented for now.